compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -8395,11 +8395,16 @@ private void addReturnIfNotPresent(BLangInvokableNode invokableNode) {
         if (invokableNode.symbol.type.getReturnType().isNullable() && (funcBody.stmts.size() < 1
                 || funcBody.stmts.get(funcBody.stmts.size() - 1).getKind() != NodeKind.RETURN)) {
             Location invPos = invokableNode.pos;
-            Location returnStmtPos = new BLangDiagnosticLocation(invPos.lineRange().filePath(),
-                    invPos.lineRange().endLine().line(),
-                    invPos.lineRange().endLine().line(),
-                    invPos.lineRange().startLine().offset(),
+            Location returnStmtPos;
+            if (invokableNode.name.value.contains(GENERATED_INIT_SUFFIX.value)) {
+                returnStmtPos = null;
```

Comment:
Seems like this area already has issues in the debugger. Need to press step in several times to step into `init` method. Giving the same position for generated init makes this case worse by automatically stepping over the init method body. Probably we can revisit here after debugger team has a proper fix for 
https://github.com/ballerina-platform/ballerina-lang/issues/30739